src/charm.py

```diff
@@ -321,7 +321,7 @@ def _on_config_changed(self, _: ConfigChangedEvent) -> None:
         state = self._setup_state()
 
         flush_runners = False
-        if state.charm_config.token != self._stored.token:
+        if self.config[TOKEN_CONFIG_NAME] != self._stored.token:
```

Comment:
this is to prevent an unnecessary flush on config change , because `charm_config.token` doesn't contain the jobmanager token value.